shell/linux/Makefile

```diff
@@ -0,0 +1,212 @@
+
+LOCAL_PATH := $(call my-dir)
+FOR_LINUX :=1
+#NO_REC := 1
+#NO_REND := 1
+WEBUI :=1 
+USE_ALSA := 1
+USE_OSS := 1
+#USE_PULSEAUDIO := 1
+USE_X11 := 1
+
+CXX=${CC_PREFIX}g++
+CC=${CC_PREFIX}gcc
+AS=${CC_PREFIX}as
+STRIP=${CC_PREFIX}strip
+LD=${CC}
+
+MFLAGS := 
+ASFLAGS := 
+LDFLAGS :=
+INCS :=
+LIBS :=
+CFLAGS := 
+CXXFLAGS :=
+
+# Platform auto-detection
+# Can be overridden by using:
+#   make platform=AMD64
+ifeq (,$(platform))
+    ARCH = $(shell uname -m)
+    ifeq ($(ARCH), $(filter $(ARCH), i386 i686))
+        platform = i686
+    else ifeq ($(ARCH), $(filter $(ARCH), x86_64 AMD64))
+        platform = AMD64
+    else ifneq (,$(findstring arm,$(ARCH)))
+        HARDWARE = $(shell grep Hardware /proc/cpuinfo)
+        ifneq (,$(findstring BCM2709,$(HARDWARE)))
+            platform = rpi2
+        else ifneq (,$(findstring AM33XX,$(HARDWARE)))
+            platform = beagle
+        else ifneq (,$(findstring Pandora,$(HARDWARE)))
+            platform = pandora
+        else
+            $(error Unsupported Hardware)
+        endif
+    else ifneq (,$(findstring mips,$(ARCH)))
+        platform = gcwz
+    else
+        $(error Unsupported CPU architecture)
+    endif
+endif
+
+$(info Platform: $(platform))
+
+# Generic x86 (32 bit)
+ifneq (,$(findstring i686,$(platform)))
+    X86_REC := 1
+    NOT_ARM := 1
+    MFLAGS += -m32
+    ASFLAGS += -32
+    LDFLAGS += -m32
+    CFLAGS += -m32 -D TARGET_LINUX_x86 -D TARGET_NO_AREC
+    CXXFLAGS += -fno-exceptions
+
+# Generic x86_64 (64 bit)
+else ifneq (,$(findstring AMD64,$(platform)))
+    X64_REC := 1
+    NOT_ARM := 1
+    CFLAGS += -D TARGET_LINUX_x86 -D TARGET_NO_AREC
+    CXXFLAGS += -fexceptions
+
+# Raspberry Pi 2
+else ifneq (,$(findstring rpi2,$(platform)))
+    AS=${CC_PREFIX}gcc
```

Comment:
Hmm, seems to be neccessary, since it won't recognize the Macros and complain about "junk at the end of the line".